### PR TITLE
[Refactor] Refactor benchmark config files

### PR DESCRIPTION
**Pull request by wangbo-zhao**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Modify the config files to adopt to the version 2.0

## Modification

Modify the config files to adopt to the version 2.0

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by wangbo-zhao** [comment]
A bug in configs/benchmarks/mmdetection/voc0712/faster_rcnn_r50_c4_mstrain_24k_voc0712.py waiting to be solved by mmdetection.

File: configs/benchmarks/mmdetection/_base_/default_runtime.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
         # dict(type='TensorboardLoggerHook')
     ])
 # yapf:enable
-custom_hooks = [dict(type='NumClassCheckHook')]
+# custom_hooks = [dict(type='NumClassCheckHook')]
```
Comment (fangyixiao18) [inline configs/benchmarks/mmdetection/_base_/default_runtime.py]: This file hasn't been updated to the 2.0 version

File: configs/benchmarks/mmdetection/coco/mask_rcnn_r50_c4_mstrain_1x_coco.py [hunk 2]
```diff
@@ -17,20 +17,26 @@
...
-data = dict(train=dict(pipeline=train_pipeline))
+train_dataloader = dict(dataset=dict(pipeline=train_pipeline))
+
+train_cfg = dict(type='EpochBasedTrainLoop', max_epochs=12, val_interval=1)
 
 custom_imports = dict(
-    imports=['tools.benchmarks.mmdetection.res_layer_extra_norm'],
+    imports=['mmselfsup.evaluation.functional.res_layer_extra_norm'],
     allow_failed_imports=False)
+
```
Comment (fangyixiao18) [inline configs/benchmarks/mmdetection/coco/mask_rcnn_r50_c4_mstrain_1x_coco.py]: The codes below should be moved to default_runtime.py, the same problem with other configs

File: configs/benchmarks/mmsegmentation/_base_/schedules/schedule_20k.py [hunk 3]
```diff
@@ -1,9 +1,24 @@
...
+        power=0.9,
+        begin=0,
+        end=20000,
+        by_epoch=False)
+]
+# training schedule for 20k
+train_cfg = dict(type='IterBasedTrainLoop', max_iters=20000, val_interval=2000)
+val_cfg = dict(type='ValLoop')
+test_cfg = dict(type='TestLoop')
+default_hooks = dict(
```
Comment (fangyixiao18) [inline configs/benchmarks/mmsegmentation/_base_/schedules/schedule_20k.py]: it is better to move default_hooks to defaulte_runtime.py

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

**Comment by fangyixiao18** [comment]
`mmdet .sh` scripts are missing?

**Comment by fangyixiao18** [comment]
`mmdet .sh` scripts are missing?

File: configs/benchmarks/mmdetection/_base_/default_runtime.py (hunk 1 above)
Comment (wangbo-zhao) [reply configs/benchmarks/mmdetection/_base_/default_runtime.py]: done

File: configs/benchmarks/mmdetection/coco/mask_rcnn_r50_c4_mstrain_1x_coco.py (hunk 2 above)
Comment (wangbo-zhao) [reply configs/benchmarks/mmdetection/coco/mask_rcnn_r50_c4_mstrain_1x_coco.py]: done

File: configs/benchmarks/mmsegmentation/_base_/schedules/schedule_20k.py (hunk 3 above)
Comment (wangbo-zhao) [reply configs/benchmarks/mmsegmentation/_base_/schedules/schedule_20k.py]: done

**Comment by wangbo-zhao** [comment]
Now all .sh scripts in mmdet and mmseg can load pretrained model without extraction.

File: configs/benchmarks/mmdetection/voc0712/faster_rcnn_r50_c4_mstrain_24k_voc0712.py [hunk 4]
```diff
@@ -11,74 +13,74 @@
...
         dict(type='TextLoggerHook', by_epoch=False),
-        # dict(type='TensorboardLoggerHook')
     ])
 
 custom_imports = dict(
-    imports=['tools.benchmarks.mmdetection.res_layer_extra_norm'],
+    imports=['mmselfsup.evaluation.functional.res_layer_extra_norm'],
     allow_failed_imports=False)
+
+env_cfg = dict(
```
Comment (fangyixiao18) [inline configs/benchmarks/mmdetection/voc0712/faster_rcnn_r50_c4_mstrain_24k_voc0712.py]: the code below can be removed, as which has been added to `default_runtime.py`

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

File: configs/benchmarks/mmdetection/voc0712/faster_rcnn_r50_c4_mstrain_24k_voc0712.py (hunk 4 above)
Comment (wangbo-zhao) [reply configs/benchmarks/mmdetection/voc0712/faster_rcnn_r50_c4_mstrain_24k_voc0712.py]: done

**Comment by fangyixiao18** [comment]
LGTM. 
@tonysy  please review the PR

**Review by fangyixiao18 — APPROVED** [review]

**Comment by wangbo-zhao** [comment]
support references config from mmdet and mmseg: done

File: configs/benchmarks/mmdetection/coco/mask_rcnn_r50_c4_mstrain_1x_coco.py [hunk 5]
```diff
@@ -1,36 +1,48 @@
...
+        init_cfg=dict(type='Pretrained', checkpoint='torchvision://resnet50')),
     roi_head=dict(
         shared_head=dict(
-            type='ResLayerExtraNorm', norm_cfg=norm_cfg, norm_eval=False)))
+            type='ResLayerExtraNorm',
+            norm_cfg=norm_cfg,
+            norm_eval=False,
+            style='pytorch')))
 
 img_norm_cfg = dict(
```
Comment (fangyixiao18) [inline configs/benchmarks/mmdetection/coco/mask_rcnn_r50_c4_mstrain_1x_coco.py]: this can be removed

**Review by tonysy — APPROVED** [review]
LGTM

File: configs/benchmarks/mmdetection/coco/mask_rcnn_r50_c4_mstrain_1x_coco.py (hunk 5 above)
Comment (fangyixiao18) [reply configs/benchmarks/mmdetection/coco/mask_rcnn_r50_c4_mstrain_1x_coco.py]: the same comments with others

Comment (wangbo-zhao) [reply configs/benchmarks/mmdetection/coco/mask_rcnn_r50_c4_mstrain_1x_coco.py]: done

**Pull request merged**